src/mscorlib/corefx/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    internal sealed partial class TlsOverPerCoreLockedStacksArrayPool<T>
+    {
+        /// <summary>Get an identifier for the current thread to use to index into the stacks.</summary>
+        private static int ExecutionId
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get
+            {
+                // On Unix, GetCurrentProcessorNumber is implemented in terms of sched_getcpu, which
```

Comment:
Yes, it is along the lines what I had in mind ... but I am not sure how well it is actually going to work in practice. It is something we should work on more (e.g. together with making GetCurrentProcessorNumber FCall). It does not need to block this change from getting merged.